connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -71,10 +71,23 @@ private static DelegatingClassLoader newDelegatingClassLoader(final List<String>
         return Utils.join(plugins, ", ");
     }
 
-    protected static <T> T newPlugin(Class<T> klass) {
+    protected <T> T newPlugin(Class<T> klass) {
+        ClassLoader classLoader = delegatingLoader.pluginClassLoader(klass.getName());
+        if (classLoader == null) {
+            classLoader = delegatingLoader;
+        }
+        return newPlugin(classLoader, klass);
+    }
+
+    protected static <T> T newPlugin(ClassLoader classLoader, Class<T> klass) {
+        ClassLoader oldClassLoader = compareAndSwapLoaders(classLoader);
+        // This doesn't use a finally block for performance reasons
```

Comment:
I wasn't aware that `finally` blocks had a large impact on performance, and haven't been able to find anything definitive on the subject that indicates they are. Can you provide a reference that backs up this claim?

Fwiw, using a `finally` block would be more concise and readable here:
```suggestion
        try {
            return Utils.newInstance(klass);
        } finally {
            compareAndSwapLoaders(oldClassLoader);
        }
```

Reply:
I would love to, and this is exactly why I left this comment.

It might be only on my laptop, but if I use a finally block here, I start getting spurious integration test timeouts that I otherwise can't explain. I refactored it to the current state, and I don't see those timeouts any more.

Reply:
So the integration tests in Connect definitely have some issues with flakily exceeding timeouts without these changes. It's strange that those failures disappear with the removal of that `finally` block; I'd be curious to see if that pattern holds when building on Jenkins. WDYT about pushing a commit with the `finally` block just to see how it builds remotely?